src/main/java/com/redhat/devtools/intellij/tektoncd/actions/ShowDiagnosticDataAction.java

```diff
@@ -14,39 +14,60 @@
 import com.intellij.openapi.ui.Messages;
 import com.redhat.devtools.intellij.common.utils.ExecHelper;
 import com.redhat.devtools.intellij.common.utils.UIHelper;
+import com.redhat.devtools.intellij.tektoncd.Constants;
 import com.redhat.devtools.intellij.tektoncd.tkn.Tkn;
 import com.redhat.devtools.intellij.tektoncd.tree.ParentableNode;
 import com.redhat.devtools.intellij.tektoncd.tree.PipelineRunNode;
 import com.redhat.devtools.intellij.tektoncd.tree.TaskRunNode;
-import java.io.IOException;
-import javax.swing.tree.TreePath;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import javax.swing.tree.TreePath;
+import java.io.IOException;
+
+import static com.redhat.devtools.intellij.tektoncd.telemetry.TelemetryService.PROP_RESOURCE_KIND;
+import static com.redhat.devtools.intellij.tektoncd.telemetry.TelemetryService.instance;
+import static com.redhat.devtools.intellij.telemetry.core.service.TelemetryMessageBuilder.ActionMessage;
+import static com.redhat.devtools.intellij.telemetry.core.util.AnonymizeUtils.anonymizeResource;
+
 public class ShowDiagnosticDataAction extends TektonAction {
     Logger logger = LoggerFactory.getLogger(ShowDiagnosticDataAction.class);
 
     public ShowDiagnosticDataAction() { super(PipelineRunNode.class, TaskRunNode.class); }
 
     @Override
     public void actionPerformed(AnActionEvent anActionEvent, TreePath path, Object selected, Tkn tkncli) {
+        ActionMessage telemetry = instance().action("show diagnostic data");
         ExecHelper.submit(() -> {
             ParentableNode element = getElement(selected);
             String namespace = element.getNamespace();
             boolean hasDataToShow = false;
             try {
                 if (element instanceof PipelineRunNode) {
+                    telemetry.property(PROP_RESOURCE_KIND, Constants.KIND_PIPELINERUN);
                     hasDataToShow = tkncli.getDiagnosticData(namespace, "tekton.dev/pipelineRun", element.getName());
                 } else if (element instanceof TaskRunNode) {
+                    telemetry.property(PROP_RESOURCE_KIND, Constants.KIND_TASKRUN);
                     hasDataToShow = tkncli.getDiagnosticData(namespace, "tekton.dev/taskRun", element.getName());
                 }
                 if (!hasDataToShow) {
+                    String errorMessage = "No data available for " + element.getName() + " in namespace " + namespace + ".";
+                    telemetry
+                            .error(anonymizeResource(element.getName(), namespace, errorMessage))
```

Comment:
Not sure this has to be considered as an error (historical data missing from cluster) maybe @lstocchi can comment

Reply:
I missed it before but yeah, you're right. This is not an actual error. It may happen there are no data

Reply:
corrected this, now sending it as #result